compose/cli/main.py

```diff
@@ -84,6 +85,20 @@ def main():
         sys.exit(1)
 
 
+def log_api_error(e):
+    if 'client is newer than server' in e.explanation:
+        # we need JSON formatted errors. In the meantime...
+        client_version = e.explanation.split('client API version: ')[1].split(',')[0]
```

Comment:
Ok fair enough, I don't see a better way forward for now (I'm just a little worried this is brittle and is going to break at some point).

Could you add `# TODO: fix this by refactoring project dispatch, docker/compose/pull/2832#commitcomment-15923800` 

After that I think it works for now.